### PR TITLE
Added system test

### LINKED ISSUE #9: enable all fitting benchmarking in a 'daily' systemtest
_From @Anders-Markvardsen on August 10, 2017 10:12_

Currently https://github.com/mantidproject/mantid/blob/master/Testing/SystemTests/tests/analysis/FittingBenchmarks.py only run a small subset of all the fitting tests it could, in order for it to be run with every pull request. 

However running all these tests would on a regular basis would provide a considerably more comprehensive testing of the Mantid fitting, e.g. catching out that for some subtle reason BFGS stopped working on some specific fitting problems #20198

Currently running the full set of test of tests takes hours. 

This issue it to add a test such as a 'skiped test' systemtest that is only run on a dedicated build, such as master_systemtests-ornl on http://builds.mantidproject.org/view/Master%20Pipeline/, on a daily or weekly basis.

_Copied from original issue: mantidproject/mantid#20218_

### PR DESCRIPTION
**Pull request by bb511**
#### Description of Work

Fixes #9 
A system test that runs daily compares the results obtained in the current build with a set of known results. Some modifications to Travis settings are required after this branch is merged.

#### Testing Instructions

No testing instructions, cron jobs should succeed on Travis url `https://travis-ci.com/mantidproject/fitbenchmarking` .

### CODE REVIEW

File: system_tests/system_test.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        nist_high_path = os.path.join(expected_results_dir, "nist_high_acc.txt")
+
+        return nist_low_path, nist_avg_path, nist_high_path
+
+    def read_expected_results_nist(self):
+
+        nist_low_path, nist_average_path, nist_high_path = \
+        self.get_expected_results_paths_nist()
+
+        with open(nist_low_path) as f:
```
Comment (AnthonyLim23) [inline system_tests/system_test.py]: can we change `f` to a more descriptive name

File: system_tests/system_test.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        expected_results_dir = self.path_to_expected_results_dir()
+        neutron_path = os.path.join(expected_results_dir, "neutron_acc.txt")
+
+        return neutron_path
+
+    def read_expected_results_neutron(self):
+
+        neutron_path = self.get_expected_results_paths_neutron()
+
+        with open(neutron_path) as f:
```
Comment (AnthonyLim23) [inline system_tests/system_test.py]: same here

File: system_tests/system_test.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+        os.path.join(nist_high_dir, "nist_higher_acc_weighted_table.txt")
+
+        return nist_low_path, nist_average_path, nist_high_path
+
+    def read_results_nist(self):
+
+        nist_low_path, nist_average_path, nist_high_path = \
+        self.get_results_paths_nist()
+
+        with open(nist_low_path) as f:
```
Comment (AnthonyLim23) [inline system_tests/system_test.py]: and here

File: system_tests/system_test.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+        return neutron
+
+
+    def test_nist(self):
+
+        expected_results = self.nist_expected_results_np_tables()
+        results = self.nist_results_np_tables()
+
+        np.testing.assert_array_almost_equal(expected_results[0], results[0], 2)
```
Comment (AnthonyLim23) [inline system_tests/system_test.py]: is 2 a bit high for the tolerance?

Comment (bb511) [reply system_tests/system_test.py]: This is the number of decimal points.

**Pull request merged**